### DISCUSSION TITLE
Enchance prepared statement by allowing positional parameters not linked to properties

### DISCUSSION THREAD

**Discussion by vic0824** [ArcadeData/arcadedb#710] [Ideas]
It seems that the sql parser in some cases does not support positional parameters that do not correspond to properties, unlike the JDBC  prepared statement, where it is possible to specify a positional parameters for each ? in the sql string.
These are the tests that I have done:

1. "SELECT FROM Order ORDER BY ? ? LIMIT ?", with parameters "processor", "DESC", "10" -> com.arcadedb.query.sql.parser.ParseException: Encountered " "?" "? "" at line 1, column 29
2. "SELECT from Order ORDER BY ? DESC LIMIT 10" with parameter "@rid" -> com.arcadedb.query.sql.parser.ParseException: Encountered " "?" "? "" at line 1, column 28
3. "SELECT FROM Order WHERE status = ? OR status = ? ORDER BY @rid ASC LIMIT ?" with parameters "READY", "PENDING", 4 (as an int) -> ok

Query 1 must therefore be re-written in this way: "SELECT FROM Order ORDER BY {sort_field} {sort_order} LIMIT ?"
and the {sort_field} and {sort_order} placeholders must be replaced with the actual values before the sql string is passed to the Database.query method.
On the other hand, Query 3 shows that in this particular case, the parser can handle a positional parameter that is not linked to a property.

Apart from the possible performance improvements, I think it would be useful to enhance the parser so that it can replace each ? with a corresponding parameter.

**Comment by lvca** [comment]
When you wrote `SELECT FROM Order ORDER BY {sort_field} {sort_order}`, did you mean passing parameters by name?

**Reply by vic0824** [reply]
It was a long time ago, but I think that's what I meant :)
